vm/systemSmartContracts/auction.go

```diff
@@ -175,25 +176,32 @@ func (s *stakingAuctionSC) unJail(args *vmcommon.ContractCallInput) vmcommon.Ret
 		return vmcommon.UserError
 	}
 
-	blsKeys, err := getBLSPublicKeys(registrationData, args)
+	blsKeys := make([][]byte, numBLSKeys)
+	for i := 0; i < len(args.Arguments); i += 2 {
+		blsKeys[i/2] = args.Arguments[i]
+	}
+
+	err = verifyBLSPublicKeys(registrationData, blsKeys)
 	if err != nil {
 		s.eei.AddReturnMessage("could not get all blsKeys from registration data: " + err.Error())
 		return vmcommon.UserError
 	}
 
-	for _, blsKey := range blsKeys {
-		vmOutput, err := s.executeOnStakingSC([]byte("unJail@" + hex.EncodeToString(blsKey)))
-		if err != nil {
-			s.eei.AddReturnMessage(err.Error())
+	transferBack := big.NewInt(0)
+	for i, blsKey := range blsKeys {
+		vmOutput, err := s.executeOnStakingSC([]byte("unJail@" + hex.EncodeToString(blsKey) + "@" + hex.EncodeToString(args.Arguments[2*i+1])))
+		if err != nil || vmOutput.ReturnCode != vmcommon.Ok {
+			transferBack.Add(transferBack, auctionConfig.UnJailPrice)
 			s.eei.Finish(blsKey)
 			s.eei.Finish([]byte{failed})
 			continue
 		}
+	}
 
-		if vmOutput.ReturnCode != vmcommon.Ok {
-			s.eei.Finish(blsKey)
-			s.eei.Finish([]byte{failed})
-		}
+	err = s.eei.Transfer(args.CallerAddr, args.RecipientAddr, transferBack, nil, 0)
```

Comment:
Is the amount that is transfered here correct? Do we want to transfer from the user to the smartContract the transferBack? Should it not be the other way back?

Reply:
the Transfer is correct, as you are transfering from the auction smart contract (args.RecipientAddr) to the caller. In Transfer the first is the destination, the second is the source.

Reply:
ok... the signature for transfer is (destination, sender) not (sender, destination)